### PR TITLE
Tempo: Upgrade @grafana/lezer-traceql patch version to use trace metrics syntax

### PR DESCRIPTION
**Pull request by joey-grafana**
**What is this feature?**

Upgrades @grafana/lezer-traceql patch version to use trace metrics syntax.

**Why do we need this feature?**

Add syntax support for trace metrics queries, introduced in [this PR](https://github.com/grafana/lezer-traceql/pull/15).

**Who is this feature for?**

Tempo users.

**Special notes for your reviewer:**

Screenshot 2024-02-15 at 11 37 28

### CODE REVIEW

**Review by adrapereira — APPROVED** [review]

**Comment by joey-grafana** [comment]
So it looks like after the updates to include metrics syntax, we exposed an existing issue that was incorrectly building the tree for `by()` and `select()` operations. Apparently they were aggregate functions because the tree was built in a way that suggested an empty `by()` or `select()` actually had a `FieldExpression` -> `IntrinsicAttribute` below it. This does not seem correct. If the operation is empty but should have args then the error should be directly below the that node in the tree.

Anyway, updated the PR to reflect a more specific error message and to also update the autocomplete for the `by` / `select` operations - i.e. continue to suggest attributes.

If you wouldn't mind taking another look @adrapereira / @fabrizio-grafana that would be great! 

Note I plan to add more specific highlighting / possibly autocomplete for the new syntax in a subsequent PR.

Before
Screenshot 2024-02-15 at 15 20 50

After
Screenshot 2024-02-15 at 15 21 00

**Comment by joey-grafana** [comment]
Ohh I should note that I'm not 100% sure why adding in the metrics syntax caused a change in how the tree is build. My guess is some kind of precedence but not certain. For context: I believe it was the [lines](https://github.com/grafana/lezer-traceql/pull/15/files#diff-852a07cd1f8351be6f1eba8618c7119968eb8f035f9952ebab8188cccde2d3c7R124-R125) with the `MetricArgs` that caused the change.

**Comment by fabrizio-grafana** [comment]
> Ohh I should note that I'm not 100% sure why adding in the metrics syntax caused a change in how the tree is build. My guess is some kind of precedence but not certain. For context: I believe it was the [lines](https://github.com/grafana/lezer-traceql/pull/15/files#diff-852a07cd1f8351be6f1eba8618c7119968eb8f035f9952ebab8188cccde2d3c7R124-R125) with the `MetricArgs` that caused the change.

Yeah probably, IIRC it happened also to me, but it shouldn't be a problem. In fact, I think we should also rethink tests in the `lezer-traceql` repo, since they check that the tree structure doesn't change, but IMO that is an implementation detail, since we don't really care about it, as long as queries are still parsed as expected. We can discuss this better live.

File: .betterer.results.json [hunk 1]
```diff
@@ -1028,6 +1028,12 @@
         "count": 5
       }
     ],
+    "/packages/grafana-ui/src/components/Splitter/Splitter.tsx": [
```
[the remark this answers is not in the document]
Comment (joey-grafana) [reply .betterer.results.json:1031]: I feel like these were left over from other PR's that I merged in from main. Not related to this PR but a side effect of making a commit. Honestly not sure why.

**Pull request merged**